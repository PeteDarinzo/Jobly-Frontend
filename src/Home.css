
.Home {
  background-color: white;
  top: 75px;
  position: relative;
  color: dimgray;
  width: 50%;
  margin: 0 auto;
  padding: 20px 50px 50px 50px;
  /* border: solid dimgray 5px; */
}

.Home h1 {
  margin: 50px;
}

.Home-left {
  text-align: left;
  color: darkslateblue;
  margin-bottom: 25px;
  font-style: italic;
}

.Home-right {
  color: darkslateblue;
  text-align: right;
  margin-top: 25px;
  font-style: italic;
}

.HomeLink {
  text-decoration: none;
  color: white;
}