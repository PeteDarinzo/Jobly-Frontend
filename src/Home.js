import React from "react";
import { Link } from "react-router-dom";
import "./Home.css";
import { Button } from "reactstrap"

const Home = ({ username }) => {
  return (
    <div className="Home shadow-lg">
      <h1>Welcome back,  {username}!</h1>
      <Link to="/jobs">
        <Button color="primary" size="lg" className="m-3">
          View Jobs
        </Button>
      </Link>
      <Link to="/companies">
        <Button color="primary" size="lg" className="m-3">
          View Companies
        </Button>
      </Link>
    </div >
  );
}

export default Home;