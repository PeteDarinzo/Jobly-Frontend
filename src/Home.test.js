import Home from "./Home";
import { render } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom"

it("renders home without crashing", () => {
  render(
    <MemoryRouter>
      <Home username="testuser" />
    </MemoryRouter>
  );
});

it("matches the home snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <Home username="testuser" />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});