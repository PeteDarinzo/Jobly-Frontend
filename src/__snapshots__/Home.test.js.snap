// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the home snapshot 1`] = `
<DocumentFragment>
  <div
    class="Home shadow-lg"
  >
    <h1>
      Welcome back, testuser!
    </h1>
    <a
      href="/jobs"
    >
      <button
        class="m-3 btn btn-primary btn-lg"
      >
        View Jobs
      </button>
    </a>
    <a
      href="/companies"
    >
      <button
        class="m-3 btn btn-primary btn-lg"
      >
        View Companies
      </button>
    </a>
  </div>
</DocumentFragment>
`;
